src/vivarium_csu_hypertension_sdc/components/hypertension_intervention.py

```diff
@@ -150,12 +155,53 @@ def measure_sbp(self, index, visit_date):
         return sbp_measurement
 
     def transition_treatment(self, index):
-        # TODO
-        pass
+        current_meds = self.population_view.subview(DOSAGE_COLUMNS + SINGLE_PILL_COLUMNS).get(index)
+
+        no_current_tx_mask = current_meds[DOSAGE_COLUMNS].sum(axis=1) == 0
+
+        if self.config.treatment_ramp == "low_and_slow":
+            if sum(no_current_tx_mask):
+                current_meds.loc[no_current_tx_mask, DOSAGE_COLUMNS] = \
+                    utilities.choose_half_dose_single_new_drug(index[no_current_tx_mask], self.med_probabilities,
+                                                               self.randomness['treatment_transition'],
+                                                               current_meds.loc[no_current_tx_mask, DOSAGE_COLUMNS])
+
+            increase_tx = index[~no_current_tx_mask]
+            min_dose_drugs_mask, min_dose_drug_in_single_pill = \
+                utilities.get_minimum_dose_drug(current_meds.loc[increase_tx])
+
+            at_max = current_meds.loc[increase_tx].mask(np.logical_not(min_dose_drugs_mask), 0).max(axis=1) == max(HYPERTENSION_DOSAGES)
+
+            # already on treatment and maxed out on dosage of minimum dose drug - add half dose of new drug
+            idx = increase_tx[at_max]
+            if not idx.empty:
+                current_meds.loc[idx, DOSAGE_COLUMNS] += \
```

Comment:
nitpick: I tend to always use parenthesis and implicit line continuation: https://www.python.org/dev/peps/pep-0008/#maximum-line-length